src/EndpointV2/Rule/RuleCreator.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Aws\EndpointV2\Rule;
+
+use Aws\Exception\UnresolvedEndpointException;
+
+class RuleCreator
+{
+    public static function create($type, $spec)
+    {
+        $validRuleTypes = ['endpoint', 'error', 'tree'];
+        if (!in_array($type, $validRuleTypes)) {
+            throw new UnresolvedEndpointException(
+                'Unknown rule type ' . $type .
+                'must be of type `Endpoint`, `Tree` or `Error`'
+            );
+        }
+
+        switch ($type) {
+            case 'endpoint':
+                return new EndpointRule($spec);
+            case 'error':
+                return new ErrorRule($spec);
+            case 'tree':
+                return new TreeRule($spec);
```

Comment:
Alternative - I think both are equally fine ways of coding this.  Instead of the check above, we can just add a default to the switch statement which throws the error.